server/verify_test.go

```diff
@@ -270,6 +348,54 @@ func TestVerifyBasicAttestation(t *testing.T) {
 	}
 }
 
+func TestVerifyBasicAttestationWithTdx(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+
+	ak, err := client.AttestationKeyRSA(rwc)
+	if err != nil {
+		t.Fatalf("failed to generate AK: %v", err)
+	}
+	defer ak.Close()
+
+	nonce := []byte("super secret nonce")
+	attestation, err := ak.Attest(client.AttestOpts{
+		Nonce: nonce,
+	})
+	if err != nil {
+		t.Fatalf("failed to attest: %v", err)
+	}
+
+	teeopts := &VerifyTdxOpts{
+		Verification: tv.DefaultOptions(),
+	}
+	if _, err := VerifyAttestation(attestation, VerifyOpts{
+		Nonce:      nonce,
+		TrustedAKs: []crypto.PublicKey{ak.PublicKey()},
+	}); err != nil {
+		t.Errorf("failed to verify: %v", err)
+	}
+
+	if _, err := VerifyAttestation(attestation, VerifyOpts{
+		Nonce:   nonce,
+		TEEOpts: teeopts,
```

Comment:
The fact that this doesn't error makes me think your test attestation doesn't include a NonHostInfo event that claims to use TDX. To avoid this kind of miss, I'd recommend that VerifyAttestation fails in VerifyTechnology when TEEOpts is non-null and the technology doesn't match the type / there is no technology event.

Reply:
I think each of NONE, SEV, and SEV_ES should not simply return nil, but also check that opts.TEEOpts is nil. That would match my expectation.